pkg/controller/cronjob/cronjob_controllerv2_test.go

```diff
@@ -409,66 +411,61 @@ func TestControllerV2SyncCronJob(t *testing.T) {
 		},
 
 		"prev ran but done, not time, A": {
-			concurrencyPolicy:          "Allow",
-			schedule:                   onTheHour,
-			deadline:                   noDead,
-			ranPreviously:              true,
-			jobCreationTime:            justAfterThePriorHour(),
-			now:                        justBeforeTheHour(),
-			expectRequeueAfter:         true,
-			expectedRequeueDuration:    1*time.Minute + nextScheduleDelta,
-			expectUpdateStatus:         true,
-			jobPresentInCJActiveStatus: true,
+			concurrencyPolicy:       "Allow",
+			schedule:                onTheHour,
+			deadline:                noDead,
+			ranPreviously:           true,
+			jobCreationTime:         justAfterThePriorHour(),
+			now:                     justBeforeTheHour(),
+			expectRequeueAfter:      true,
+			expectedRequeueDuration: 1*time.Minute + nextScheduleDelta,
+			expectUpdateStatus:      true,
 		},
 		"prev ran but done, not time, F": {
-			concurrencyPolicy:          "Forbid",
-			schedule:                   onTheHour,
-			deadline:                   noDead,
-			ranPreviously:              true,
-			jobCreationTime:            justAfterThePriorHour(),
-			now:                        justBeforeTheHour(),
-			expectRequeueAfter:         true,
-			expectedRequeueDuration:    1*time.Minute + nextScheduleDelta,
-			expectUpdateStatus:         true,
-			jobPresentInCJActiveStatus: true,
```

Comment:
I don't think you should be removing this pre-condition, but given we expect status update, then we also request a completion of the active job. In which case I'd add `expectCompleted: true`, rather than dropping `jobPresentInCJActiveStatus: true`. I've tried a few cases of yours, but it seems this should apply to most where the two conditions are set.

Reply:
Added the `expectCompleted: true` in and revertet the removal of `jobPresentInCJActiveStatus: true`